File: src/main/java/org/bakkes/game/view/battle/PokeView.java
package org.bakkes.game.view.battle;

import java.nio.file.Path;

import org.bakkes.game.model.GameInfo;
import org.bakkes.game.model.pokemon.Pokemon;
import org.bakkes.game.view.IRenderable;
import org.bakkes.game.view.SpriteType;
import org.bakkes.game.view.components.Sprite;
import org.newdawn.slick.Graphics;

import com.google.inject.Inject;
import com.google.inject.name.Named;

public class PokeView implements IRenderable{
    private @Inject Sprite avatar;
    private @Inject StatisticsView stats;
    private @Inject @Named("spritePokemon") Path path;

	public void setPokemon(final Pokemon to){
		avatar.setSpritePath(path.resolve(SpriteType.big.getImageName(to)));
		stats.setPokemon(to);
		final float margin = 40;
		stats.x(GameInfo.SCREEN_WIDTH - stats.width() - margin);
		stats.y(margin);
	}

	@Override
	public void render(final Graphics g) {
		avatar.render(g);
		stats.render(g);
	}


}

File: src/main/scala/org/bakkes/game/view/overworld/PokeView.scala
package org.bakkes.game.view.overworld

import java.nio.file.Path

import com.google.inject.Inject
import com.google.inject.name.Named
import org.bakkes.game.model.GameInfo
import org.bakkes.game.model.pokemon.Pokemon
import org.bakkes.game.view.{IRenderable, PokemonSprite}
import org.bakkes.game.view.SpriteType._
import org.bakkes.game.view.battle.StatisticsView
import org.newdawn.slick.Graphics

class PokeView @Inject() (
	private val avatar: PokemonSprite,
	private val stats: StatisticsView,
	@Named("spritePokemon")
	private val path: Path
) extends IRenderable {

	def setPokemon(to: Pokemon) {
		avatar.spriteType = big
		avatar.setPokemon(to)
		stats.setPokemon(to)
		val margin: Float = 40
		stats.x(GameInfo.SCREEN_WIDTH - stats.width - margin)
		stats.y(margin)
	}

	def render(g: Graphics) {
		avatar.render(g)
		stats.render(g)
	}
}